### Treść
Za siedmioma górami, za siedmioma pustyniami, w królestwie zwierzżt, król Skaza i kandydat na króla – Simba, biorą udział w pisemnym konkursie.  Podług bardzo rzetelnego obserwatora tj. Drzewa Życia, Skaza powinien dostać 49 punktów, a Simba – 51. Niestety, obserwator nie jest obywatelem królestwa zwierząt i nie może wybierać za nich. Dlatego o wygraniu konkursu mają zadecydować srednie oceny kandydatów dokonanywane przez zwierzęcych wyborców. Wygrywa ten kandydat, który osiągnie większą srednią ocen wśród wyborców. Wiadomo, że 15% zwierząt jest inteligentne. Podobnie, 15% zwierząt jest nierozsądnych. Pozostałe zwierzęta nazwijmy przeciętnymi. Wiadomo, ze wariancje ocen dokonywanych przez zwierzęcych wyborców róznią się tylko ze względu na powyższą klasyfikację inteligencji. Dla inteligentnego zwierzęcia, błąd popełniany względem oceny Drzewa życia to zmienna losowa o rozkładzie zgodnym z X, dla przeciętnego – ze zmienną Y, a nierozsądnego – ze zmienną Z. Zakładamy, że X, Y, Z mają niezalezne rozkłady normalne, o wspólnej średniej 0, oraz wariancjach: $Var(X) = 1$, $Var(Y) = 4$ i $Var(Z) = 9$, a ponadto wiemy, ze wszystkie zwierzęta kazdą ocenę przeprowadzają niezależnie od innych ocen (tak że tych innych zwierząt). Początkowo tylko 10 zwierząt wie o wyborach i planuje oceniać prace kandydatów. Są wsród nich 2 zwierzęta inteligentne, 6 przeciętnych i 2 nierozsądne.

a) Niech $N_1 = N(0, \sigma_1), N_2 = N(0, \sigma_2)$, będą niezaleznymi zmiennymi losowymi. Jaki rozkład ma $N_1 + N_2$?

Z własności rozkładu normalnego $N_1 + N_2 = N(\mu_1 + \mu_2, \sigma_1 + \sigma_2) = N(0, \sigma_1 + \sigma_2)$

b) Jakie rozkłady mają zmienne opisujące sumy ocen od 10 początkowych wyborców (liczone osobno dla Skazy i Simby)? 
A jaki rozkład ma różnica tych sum?
Jaki rozkład ma różnica srednich ocen dla poszczególnych kandydatów?

Z faktu z a) mamy:

$N_{Skaza} = N(49 * 10, 1 * 2 + 4 * 6 + 2 * 9) = N(490, 44)$

$N_{Simba} = N(51 * 10, 1 * 2 + 4 * 6 + 2 * 9) = N(510, 44)$

Ustalmy $R = N_{Simba} - N_{Skaza} = N_{Simba} + (-N_{Skaza})$

Jako, że $N_{Skaza}$ ma rozkład normalny, to po negacji wariancja się nie zmienia: $-N_{Skaza} \sim N(-490,44)$, czyli dalej z faktu a)

$N_{Simba} + (-N_{Skaza}) = N(20, 88)$

Średnie ocen to oczywiście $A_{Simba} = N_{Simba}/10$, $A_{Skaza} = N_{Skaza}/10$. Dalej mamy

$A_{Simba} \sim N(\frac{1}{10} \cdot 510, (\frac{1}{10})^2 \cdot 44) = (51, 0.44)$
$A_{Skaza} \sim N(\frac{1}{10} \cdot 490, (\frac{1}{10})^2 \cdot 44) = (49, 0.44)$

I w końcu

$N_{Simba} - N_{Skaza} = N(2,0.88)$

c) Jakie jest prawdopodobieństwo, ze przy wskazanych 10 wyborcach, Skaza wygra wybory (będzie miał większą srednią ocen)?

Wskazówka: Znormalizuj odpowiednią zmienną i skorzystaj z tablicy dystrybuanty standardowego rozkładu normalnego $N(0, 1)$.

Prawdopodobieństwo zwycięzcy Skazy to $P(S_{Simba} - S_{Skaza} < 0) = P(R < 0)$

Wprowadźmy $R' = aR + b, R' \sim N(0,1)$, znormalizowaną zmienną $R$. Policzmy $a$ i $b$.

$E[aR + b] = 0 \implies aE[R] + b = 0 \implies b = -aE[R]$

$Var[aR + b] = 1 \implies a^2Var[R] = 1 \implies a = \frac{1}{\sqrt{Var[R]}}$

$a = \frac{1}{\sqrt{88}} = \frac{1}{2\sqrt{22}}$

$b = \frac{20}{2\sqrt{22}} = \frac{10}{\sqrt{22}}$

$P(R < 0) = P(\frac{R' - b}{a} < 0) = P(R' - b < 0) = P(R' < b) = P(R' < -\frac{10}{\sqrt{22}}) \approx P(R' < -2.132)$

Z tablic mamy, że $P(X < -2.13) = 0.01659$

d) Kapituła konkursowa rozwaza kilka strategii reklamy wyborów, aby zwiększyć liczbę wyborców. Liczbę nowych wyborców danego typu w zależnoci od wybranego typu reklamy, przedstawia poniższa tabela:
$$
\begin{array}{c|c|c|c|c}
\text{Reklama} & \text{brak} & \text{hieny} & \text{poczta Zazu} & \text{zebranie u Rafikiego}\\\hline\hline
\text{Inteligentne} & +0 & +0 & +3 & +9\\
\text{Przeciętne} & +0 & +10 & +12 & +1\\
\text{Nierozsądne} & +0 & +10 & +5 & +0\\
\end{array}
$$
Ponadto Skaza ma dwie strategie: przekonać 5 nowych nierozsądnych wyborców albo nie mówić nikomu więcej o
wyborach.
Wypłata Skazy to prawdopodobieństwo wygrania konkursu przez niego, a Kapituły (reprezentującej całe królestwo
zwierząt) – prawdopodobieństwo wyborów zgodnych z rzetelną oceną Drzewa Zycia (czyli wygranej Simby).
Przedstaw grę między Skazą, a Kapitułą w postaci macierzowej (jedna komórka wynika z poprzedniego podpunktu).

Macież liczby wyborców:

$$
\begin{array}{c|c|c|c|c}
\text{Skaza}\backslash\text{Kapituła} & \text{brak} & \text{Hieny} & \text{Zazu} & \text{Rafiki}\\\hline
\text{cisza} & (2,6,2) & (2,16,12) & (5,18,7) & (11,7,2)\\
\text{manipulacja} & (2,6,7) & (2,16,17) & (5,18,12) & (11,7,7)
\end{array}
$$

In [39]:
import math
from scipy.stats import norm

def calcProb(I, P, N):
    Simba = (51 * sum([I,P,N]), I + 4 * P + 9 * N)
    Skaza = (49 * sum([I,P,N]), I + 4 * P + 9 * N)

    R = (Simba[0] - Skaza[0], Simba[1] + Skaza[1])

    a = 1/math.sqrt(R[1])
    b = -a * R[0]

    return norm.cdf(b)

voters = [(2,6,2),
          (2,6,7),
          (2,16,12),
          (2,16,17),
          (5,18,7),
          (5,18,12),
          (11,7,2),
          (11,7,7)]

for v in voters:
    p = calcProb(v[0], v[1], v[2])
    print('%.8f'%(p), '%.8f'%(1-p))

0.01650313 0.98349687
0.01226914 0.98773086
0.00064919 0.99935081
0.00041181 0.99958819
0.00016810 0.99983190
0.00013678 0.99986322
0.00008972 0.99991028
0.00023202 0.99976798


$$
\begin{array}{c|c|c|c|c}
\frac{\text{Skaza}}{\text{Kapituła}} & \text{brak} & \text{Hieny} & \text{Zazu} & \text{Rafiki}\\\hline\\[1pt]
\text{cisza} & \frac{0.01650313}{0.98349687} & \frac{0.00064919}{0.99935081} & \frac{0.00016810}{0.99983190} & \frac{0.00008972}{0.99991028}\\[10pt]\hline\\[1pt]
\text{manipulacja} & \frac{0.01226914}{0.98773086} & \frac{0.00041181}{0.99958819} & \frac{0.00013678}{0.99986322} & \frac{0.00023202}{0.99976798}\\[10pt]
\end{array}
$$

e) Znajdź równowagi Nasha dla tej gry.

Nie ma takich, niezależnie od której strategii rozpoczniemy, odpowiadjąc na nie wpadniemy w cykl:

$Brak \rightarrow Cisza \rightarrow Rafiki \rightarrow Manipulacja \rightarrow Zazu \rightarrow Cisza \rightarrow \dots$

$Hieny \rightarrow Cisza \rightarrow \dots$

f) Co charakteryzuje grupę, która zwiększa uczciwość wyborców?

Jeśli za uczciwe wybory przyjmiemy te zgodne z osądem Drzewa Życia, to grupą zwiększającą uczciwość są zwierzęta intelignetne, ze względu na małą wariancję swoich wyborów.